src/Features/Core/Portable/CodeStyle/AbstractCodeStyleProvider.Analysis.cs

```diff
@@ -0,0 +1,139 @@
+            {
+                _codeStyleProvider = codeStyleProvider;
+                _context = context;
+            }
+
+            public void RegisterCompilationStartAction(Action<Compilation, AnalysisContext> analyze)
+            {
+                var _this = this;
+                _context.RegisterCompilationStartAction(
+                    c => analyze(c.Compilation, _this));
+            }
+
+            public void RegisterCodeBlockAction(Action<CodeBlockAnalysisContext, CodeStyleOption<TOptionKind>> analyze)
+            {
+                var provider = _codeStyleProvider;
+                _context.RegisterCodeBlockAction(
+                    c => AnalyzeIfEnabled(provider, c, analyze, c.Options, c.SemanticModel.SyntaxTree, c.CancellationToken));
+            }
+
+            public void RegisterSemanticModelAction(Action<SemanticModelAnalysisContext, CodeStyleOption<TOptionKind>> analyze)
+            {
+                var provider = _codeStyleProvider;
+                _context.RegisterSemanticModelAction(
+                    c => AnalyzeIfEnabled(provider, c, analyze, c.Options, c.SemanticModel.SyntaxTree, c.CancellationToken));
+            }
+
+            public void RegisterSsyntaxTreeAction(Action<SyntaxTreeAnalysisContext, CodeStyleOption<TOptionKind>> analyze)
+            {
+                var provider = _codeStyleProvider;
+                _context.RegisterSyntaxTreeAction(
+                    c => AnalyzeIfEnabled(provider, c, analyze, c.Options, c.Tree, c.CancellationToken));
+            }
+
+            public void RegisterOperationAction(
+                Action<OperationAnalysisContext, CodeStyleOption<TOptionKind>> analyze,
+                params OperationKind[] operationKinds)
+            {
+                var provider = _codeStyleProvider;
+                _context.RegisterOperationAction(
+                    c => AnalyzeIfEnabled(provider, c, analyze, c.Options, c.Operation.SemanticModel.SyntaxTree, c.CancellationToken),
+                    operationKinds);
+            }
+
+            public void RegisterSyntaxNodeAction(
+                Action<SyntaxNodeAnalysisContext, CodeStyleOption<TOptionKind>> analyze,
+                params TSyntaxKind[] syntaxKinds)
+            {
+                var provider = _codeStyleProvider;
+                _context.RegisterSyntaxNodeAction(
+                    c => AnalyzeIfEnabled(provider, c, analyze, c.Options, c.SemanticModel.SyntaxTree, c.CancellationToken),
+                    syntaxKinds);
+            }
+
+            private static void AnalyzeIfEnabled<TContext>(
+                TCodeStyleProvider provider, TContext context, Action<TContext, CodeStyleOption<TOptionKind>> analyze,
+                AnalyzerOptions options, SyntaxTree syntaxTree, CancellationToken cancellationToken)
+            {
+                var optionSet = options.GetDocumentOptionSetAsync(syntaxTree, cancellationToken).GetAwaiter().GetResult();
```

Comment:
Yes.  I would be fine with the engine computing this.  Let me know when that is done and we can remove some of this code :)

Reply:
This implementation aligns with what analyzers do today, so it's not inherently a problem. I wouldn't spend too much time changing it to something else considering it will be replaced by the new .editorconfig-in-compiler functionality.